pkg/devfile/adapters/common/command.go

```diff
@@ -134,17 +147,32 @@ func GetRunCommand(data data.DevfileData, devfileRunCmd string) (runCommand comm
 // ValidateAndGetPushDevfileCommands validates the build and the run command,
 // if provided through odo push or else checks the devfile for devBuild and devRun.
 // It returns the build and run commands if its validated successfully, error otherwise.
-func ValidateAndGetPushDevfileCommands(data data.DevfileData, devfileBuildCmd, devfileRunCmd string) (pushDevfileCommands []common.DevfileCommand, err error) {
+func ValidateAndGetPushDevfileCommands(data data.DevfileData, devfileInitCmd, devfileBuildCmd, devfileRunCmd string) (pushDevfileCommands []common.DevfileCommand, err error) {
 	var emptyCommand common.DevfileCommand
-	isBuildCommandValid, isRunCommandValid := false, false
+	isInitCommandValid, isBuildCommandValid, isRunCommandValid := false, false, false
+
+	initCommand, initCmdErr := GetInitCommand(data, devfileInitCmd)
+
+	isInitCmdEmpty := reflect.DeepEqual(emptyCommand, initCommand)
+	if isInitCmdEmpty && initCmdErr == nil {
+		// If there was no init command specified through odo push and no default init command in the devfile, default validate to true since the init command is optional
+		isInitCommandValid = true
+		glog.V(3).Infof("No init command was provided")
+	} else if !isInitCmdEmpty && initCmdErr == nil {
+		isInitCommandValid = true
+		pushDevfileCommands = append(pushDevfileCommands, initCommand)
+		glog.V(3).Infof("Init command: %v", initCommand.Name)
+	}
 
 	buildCommand, buildCmdErr := GetBuildCommand(data, devfileBuildCmd)
```

Comment:
can we please add check for `buildCmdErr`? this would simplify the `isBuildCmdEmpty && buildCmdErr == nil` statement and make it more readable.

Reply:
Same as the `initCmdErr`, we check if `buildCmdErr` is nil further down in the flow. It would be possible to refactor the code to accommodate this though, to make it a bit cleaner.